packages/xstate-store/src/store.ts

```diff
@@ -162,6 +199,144 @@ function createStoreCore<
   return store;
 }
 
+export type TransitionsFromEventPayloadMap<
+  TEventPayloadMap extends EventPayloadMap,
+  TContext extends StoreContext,
+  TEmitted extends EventObject
+> = {
+  [K in keyof TEventPayloadMap & string]:
+    | StoreAssigner<
+        NoInfer<TContext>,
+        {
+          type: K;
+        } & TEventPayloadMap[K],
+        TEmitted
+      >
+    | StorePropertyAssigner<
+        NoInfer<TContext>,
+        {
+          type: K;
+        } & TEventPayloadMap[K],
+        TEmitted
+      >;
+};
+
+/**
+ * Creates a **store** that has its own internal state and can be sent events
+ * that update its internal state based on transitions.
+ *
+ * @example
+ *
+ * ```ts
+ * const store = createStore({
+ *   types: {
+ *     // ...
+ *   },
+ *   context: { count: 0 },
+ *   on: {
+ *     inc: (context, event: { by: number }) => {
+ *       return {
+ *         count: context.count + event.by
+ *       };
+ *     }
+ *   }
+ * });
+ *
+ * store.subscribe((snapshot) => {
+ *   console.log(snapshot);
+ * });
+ *
+ * store.send({ type: 'inc', by: 5 });
+ * // Logs { context: { count: 5 }, status: 'active', ... }
+ * ```
+ */
+export function createStore<
```

Comment:
why do we need **four** overloads here? this is soemwhat surprising

Reply:
I couldn't figure out "either `types` or `schema`, or none of them" and have it infer correctly. Maybe you can?